web/packages/teleport/src/AppLauncher/AppLauncher.tsx

```diff
@@ -140,6 +146,19 @@ export function AppLauncherAccessDenied(props: AppLauncherAccessDeniedProps) {
   return <AccessDenied message={props.statusText} />;
 }
 
+// stripFQDN removes the port from the FQDN if it has one and ensures
+// the FQDN is lowercase. This is to prevent issues matching the
+// resolved fqdn with the one that was passed. Apps generally aren't
+// supposed to have a port in the public address but some integrations
+// create apps that do.
+function stripFQDN(fqdn: string) {
+  const portStartIdx = fqdn.indexOf(':');
```

Comment:
Is it possible for an app's FQDN to be an IPv6 address? I don't think this logic would work on an IPv6 address.

Reply:
That's a great question, I'm not sure. I know that apps created from a static config are explicitly not allowed to have IP addresses in the `public_addr` (https://github.com/gravitational/teleport/blob/master/lib/service/servicecfg/app.go#L120) but that check is not enforced for dynamically created apps. I'm pretty sure that's a bug, but I know other app servers (such as the AWS app server) seem to always create apps with a port, which is also disallowed for statically created apps but not dynamically created ones.

I would guess IP addresses aren't allowed in app public addresses, but I'll confirm to be sure.

Reply:
IP addresses have never historically been allowed in app `public_addrs` AFAIK.